src/components/insync/text-field/textField.tsx

```diff
@@ -0,0 +1,172 @@
+		<TextFieldStyled isDark={isDark}>
+			<label className="label">{label}</label>
+			<div
+				className={`input-wrapper ${isHovered ? 'hovered' : ''} ${isActive ? 'active' : ''} ${error ? 'error' : ''} ${
+					disabled ? 'disabled' : ''
+				}`}>
+				<span className={`dot ${disabled ? 'disabled' : ''}`} />
+				<input
+					type="text"
+					onMouseEnter={handleMouseEnter}
+					onMouseLeave={handleMouseLeave}
+					onMouseDown={handleMouseDown}
+					onMouseUp={handleMouseUp}
+					disabled={disabled}
+				/>
+				<span className={`dot ${disabled ? 'disabled' : ''}`} />
+				<button className={`max-button ${disabled ? 'disabled' : ''}`} disabled={disabled}>
+					Max
+				</button>
+			</div>
+			<div className="assistive-text">{assistiveText}</div>
+			{error && <div className="error-message"> {errorMessage} </div>}
+		</TextFieldStyled>
+	);
+};
+
+const TextFieldStyled = styled.div<StyledProps>`
+	display: flex;
+	flex-direction: column;
+	align-items: left;
+	width: fill;
+	height: 48px;
+	font-family: Inter, ui-sans-serif, system-ui;
+	margin-bottom: 50px;
+
+	.label {
+		font-size: 14px;
+		margin-bottom: 4px;
+		font-weight: 500;
+		color: var(--flip-color);
+	}
+
+	.assistive-text {
+		font-size: 12px;
+		margin-top: 4px;
+		font-weight: 400;
+		color: var(--8-2-color);
+	}
+
+	.error-message {
+		font-size: 12px;
+		margin-top: 4px;
+		font-weight: 400;
+		color: ${colors.error};
+	}
+
+	.dot {
+		height: 24px;
+		width: 24px;
+		background-color: transparent;
+		border: 2px solid var(--flip-color);
+		border-radius: 50%;
+		display: inline-block;
+
+		&.disabled {
+			border: 2px solid var(--6-4-color);
+			pointer-events: none;
+		}
+	}
+
+	.input-wrapper {
+		display: flex;
+		align-items: center;
+		border: 1px solid var(--4-6-color);
+		border-radius: 8px;
+		padding: 8px;
+		gap: 8px;
+		background-color: var(--background-color);
+		color: ${colors.gray4};
+
+		&.hovered {
+			border-color: var(--6-4-color);
+		}
+
+		&.active {
+			border: 2px solid;
+			border-image-slice: 1;
+			border-image-source: ${colors.primary};
+		}
+
+		&.disabled {
+			background-color: var(--2-8-color);
+			pointer-events: none;
+		}
+
+		&.error {
+			border-color: ${colors.error};
+		}
+
+		input {
+			flex: 1;
+			background-color: transparent;
+			border: none;
+			outline: none;
+			color: var(--6-4-color);
+			position: relative;
+		}
+
+		.max-button {
+			background-color: var(--2-8-color);
+			color: var(--flip-color);
+			border: none;
+			padding: 6px 16px;
+			border-radius: 8px;
+			cursor: pointer;
+
+			&.disabled {
+				background-color: var(--4-6-color);
+				color: var(--6-4-color);
+				pointer-events: none;
+			}
+		}
+	}
+
+	--background-color: ${props => (props.isDark ? colors.deepBlack : colors.white)};
+	--flip-color: ${props => (props.isDark ? colors.white : colors.trueBlack)};
+	--8-2-color: ${props => (props.isDark ? colors.gray8 : colors.gray2)};
+	--2-8-color: ${props => (props.isDark ? colors.gray2 : colors.gray8)};
```

Comment:
We should definately not be doing this, you should be having one property which changes values based on which theme we are using, such as primaryColor instead of gray2 and gray8